onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,116 @@
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write('User {} deleted successfully.'.format(username))
+        # confirm too that no user exists with provided email
+        try:
+            User.objects.get(email=user.email).delete()
+        except User.DoesNotExist:
+            pass
+    except User.DoesNotExist:
+        sys.stdout.write('User {} does not exist.'.format(username))
+
+
+class Command(BaseCommand):
+    """
+    Delete users management command.
+    """
+    help = 'Delete users'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user', nargs='*')
+
+        parser.add_argument(
+            '--user_input',
+            action='store_true',
+            help='Confirm deletion of user account',
+            default=False
+        )
+
+    def handle(self, *args, **kwargs):
+        users = kwargs.get('user')
+        user_input = kwargs.get('user_input')
+
+        if users:
+            for user in users:
+                user_details = user.split(':')
+                username = user_details[0]
+                email = user_details[1]
+                if user_input is False:
+                    # If the --user_input flag is not provided.
+                    user_stats = get_user_object_stats(username)
+
+                    # Get acknowledgement from the user on this
+                    user_input = input(
+                        "User account '{}' has {} projects, {} forms and {} submissions. \
+                        Do you wish to continue deleting this account?".format(
+                            username,
+                            user_stats['projects'],
+                            user_stats['forms'],
+                            user_stats['submissions']
+                            ))
+                    if user_input in ["Yes", "Y", "yes", "y"]:
+                        inactivate_user(username)
+                        sys.stdout.write(
+                            "User Account {} deleted successfully!".format(
+                                username))
+
+                        # confirm too that no user exists with provided email
+                        try:
+                            User.objects.get(email=email).delete()
+                        except User.DoesNotExist:
+                            sys.stdout.write(
+                                "No User Account for the given email.")
```

Comment:
Doesn't `inactivate_user` already do this ?

Reply:
Removed this.